Libraries/LibJS/Runtime/Utf16String.cpp

```diff
@@ -34,15 +34,20 @@ NonnullRefPtr<Utf16StringImpl> Utf16StringImpl::create(Utf16Data string)
 
 NonnullRefPtr<Utf16StringImpl> Utf16StringImpl::create(StringView string)
 {
-    return create(MUST(utf8_to_utf16(string)));
+    auto result = MUST(utf8_to_utf16(string));
+    auto impl = create(move(result.data));
+    impl->m_cached_view.unsafe_set_code_point_length(result.code_point_count);
+    return impl;
 }
 
 NonnullRefPtr<Utf16StringImpl> Utf16StringImpl::create(Utf16View const& view)
 {
     Utf16Data string;
     string.ensure_capacity(view.length_in_code_units());
     string.unchecked_append(view.data(), view.length_in_code_units());
-    return create(move(string));
+    auto impl = create(move(string));
+    impl->m_cached_view = view;
```

Comment:
er. what? isn't this stack use after scope? Why aren't you setting the cached view to a view of the *new* string you just created?

Reply:
Fixed, hopefully this was the source of the UAF too.